server/database.go

```diff
@@ -943,3 +950,138 @@ func validateDatabaseLabelList(labelList []*api.DatabaseLabel, labelKeyList []*a
+			Tp:       ast.TableOptionEngine,
+			StrValue: createTableContext.Engine,
+		})
+	}
+	if createTableContext.CharacterSet != "" {
+		createTableOptions = append(createTableOptions, &ast.TableOption{
+			Tp:       ast.TableOptionCharset,
+			StrValue: createTableContext.CharacterSet,
+		})
+	}
+	if createTableContext.Collation != "" {
+		createTableOptions = append(createTableOptions, &ast.TableOption{
+			Tp:       ast.TableOptionCollate,
+			StrValue: createTableContext.Collation,
+		})
+	}
+	if createTableContext.Comment != "" {
+		createTableOptions = append(createTableOptions, &ast.TableOption{
+			Tp:       ast.TableOptionComment,
+			StrValue: createTableContext.Comment,
+		})
+	}
+	createTableStmt.Options = createTableOptions
+
+	columnDefs := []*ast.ColumnDef{}
+	for _, addColumnContext := range createTableContext.AddColumnList {
+		columnDef := transformAddColumnContext(addColumnContext, engineType)
+		columnDefs = append(columnDefs, columnDef)
+	}
+	createTableStmt.Cols = columnDefs
+
+	return createTableStmt
+}
+
+func transformAddColumnContext(addColumnContext *api.AddColumnContext, engineType db.Type) *ast.ColumnDef {
+	colName := &ast.ColumnName{
+		Name: model.NewCIStr(addColumnContext.Name),
+	}
+	columnDef := &ast.ColumnDef{
+		Name: colName,
+	}
+
+	var colType *types.FieldType
+	if engineType == db.MySQL {
+		colType = types.NewFieldType(transformColumnTypeForMySQL(addColumnContext.Type))
+	} else {
+		types.DefaultTypeForValue(interface{}(addColumnContext.Type), colType, addColumnContext.CharacterSet, addColumnContext.Collation)
+	}
+	columnDef.Tp = colType
+
+	columnOptionList := []*ast.ColumnOption{}
+	if addColumnContext.Comment != "" {
+		columnOptionList = append(columnOptionList, &ast.ColumnOption{
+			Tp:   ast.ColumnOptionComment,
+			Expr: ast.NewValueExpr(interface{}(addColumnContext.Comment), addColumnContext.CharacterSet, addColumnContext.Collation),
+		})
+	}
+	if addColumnContext.Collation != "" {
+		columnOptionList = append(columnOptionList, &ast.ColumnOption{
+			Tp:       ast.ColumnOptionCollate,
+			StrValue: addColumnContext.Collation,
+		})
+	}
+	if addColumnContext.Default != nil {
+		columnOptionList = append(columnOptionList, &ast.ColumnOption{
+			Tp:   ast.ColumnOptionDefaultValue,
+			Expr: ast.NewValueExpr(interface{}(*addColumnContext.Default), addColumnContext.CharacterSet, addColumnContext.Collation),
+		})
+	}
+	if !addColumnContext.Nullable {
+		columnOptionList = append(columnOptionList, &ast.ColumnOption{
+			Tp: ast.ColumnOptionNotNull,
+		})
+	}
+	columnDef.Options = columnOptionList
+
+	return columnDef
+}
+
+func transformColumnTypeForMySQL(typeStr string) byte {
+	switch typeStr {
+	// Maybe it should be a regexp to match like `varchar(%d+)`.
+	case "varchar":
+		return mysql.TypeVarchar
+	case "int":
+		return mysql.TypeLong
+	case "bigint":
+		return mysql.TypeLonglong
+	case "json":
+		return mysql.TypeJSON
+	}
+
+	return mysql.TypeUnspecified
```

Comment:
A bit worried about the completeness of this type conversion.

Reply:
This PR is mainly introducing the way of restoring `DatabaseEdit` to DDL. And I'd like to refine these type definition in a separate PR. WDYT?

Reply:
Yes, exactly. Let's tighten this area in future PRs.